### DISCUSSION TITLE
Deploy Netbox to K8s

### DISCUSSION THREAD

**Discussion by sonbm7112** [netbox-community/netbox#14519] [General]
Hi all, I want to deploy Netbox to K8s, please help me know:
- What needs to be done
- Conditions to ensure implementation
- Advantages and disadvantages compared to pure Docker deployment on traditional server

Thanks!!

**Comment by candlerb** [comment]
You'll need to write Kubernetes manifests to do everything the docker-compose files do: deploy the Netbox front-end server(s) and the Netbox background tasks. You'll need to run postgres and redis (either using third-party manifests, or writing those yourself too) and get Netbox to talk to them.

You will also need to provide access to shared storage for media and scripts. This can be done "cloud natively" if you provide an S3 type storage service in your cluster. Or, you could have something like a shared read-write NFS volume mount. If you restrict yourself to a single front-end server pod then a local stateful volume may be sufficient.

You'll have to configure an ingress controller for web requests to be forwarded to Netbox. You'll have to decide whether to do HTTPS termination on the ingress and forward HTTP (which I think is typical), or to TCP-forward to Netbox and do the HTTPS termination there. You'll need to sort out HTTPS certificates, e.g. using cert-manager.

Advantages: you can scale and replicate this, e.g. if you want to run 100 separate Netbox instances for different customers, you can fire them all up in 100 different Kubernetes namespaces using essentially the same manifests.

Disadvantages: you'll have a ton of work to write all those manifests and get everything talking together successfully.

If I google "netbox kubernetes" I find https://github.com/CENGN/netbox-kubernetes - you could try this as a starting point, but as it hasn't had any commits for a year, and says it was tested with Netbox 2.11.12, you may have a lot of work to bring it up to date.

FWIW, I did once try to use `kompose` to convert docker-compose files into Kubernetes manifests automatically - but for LibreNMS, not Netbox. My experience is at https://github.com/kubernetes/kompose/issues/1390. In short, there may be a lot of fixing up to do if you go this way too.

But if you just want to deploy a single Netbox server instance, then your life will be much easier if you just run a VM or docker-compose.

**Comment by abhi1693** [comment]
You can also look into https://github.com/bootc/netbox-chart

**Comment by bitcollector1** [comment]
Thanks for the pointers folks. I'm running suzieq on K8 but have not went down that path just yet for NetBox.  It's nice to have all the steps in case I decide to go down that path at some point.
